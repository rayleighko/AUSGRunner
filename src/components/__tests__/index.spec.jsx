import React from 'react'
import {shallow} from 'enzyme'
import {
	Layout, StatusBar, LogPanel, CreatedBy, Home, PlayGround, 
	BackToHome, Player, Ball, Map, BottomTile, WallTile
} from '../'


describe('Common Components', () => {
	describe('Layout test suite', () => {
		it('should render', () => {
			const tree = shallow(<Layout/>)
			expect(tree).toMatchSnapshot()
		})
	})

	describe('StatusBar test suite', () => {
		it('should render', () => {
			const tree = shallow(<StatusBar/>)
			expect(tree).toMatchSnapshot()
		})
	})

	describe('LogPanel test suite', () => {
		it('should render', () => {
			const tree = shallow(<LogPanel/>)
			expect(tree).toMatchSnapshot()
		})
	})
})

describe('Menu Components', () => {
	describe('CreatedBy test suite', () => {
		it('should render', () => {
			const tree = shallow(<CreatedBy/>)
			expect(tree).toMatchSnapshot()
		})
	})

	describe('Home test suite', () => {
		it('should render', () => {
			const tree = shallow(<Home/>)
			expect(tree).toMatchSnapshot()
		})
	})

	describe('PlayGround test suite', () => {
		it('should render', () => {
			const tree = shallow(<PlayGround/>)
			expect(tree).toMatchSnapshot()
		})
	})

	describe('BackToHome test suite', () => {
		it('should render', () => {
			const tree = shallow(<BackToHome/>)
			expect(tree).toMatchSnapshot()
		})
	})
})

describe('Creature Components', () => {
	describe('Player test suite', () => {
		it('should render', () => {
			const tree = shallow(<Player/>)
			expect(tree).toMatchSnapshot()
		})
	})
})

describe('Elements Components', () => {
	describe('Ball test suite', () => {
		it('should render', () => {
			const tree = shallow(<Ball/>)
			expect(tree).toMatchSnapshot()
		})
	})
})

describe('Map Components', () => {
	describe('Map test suite', () => {
		it('should render', () => {
			const tree = shallow(<Map/>)
			expect(tree).toMatchSnapshot()
		})
	})

	describe('BottomTile test suite', () => {
		it('should render', () => {
			const tree = shallow(<BottomTile/>)
			expect(tree).toMatchSnapshot()
		})
	})

	describe('WallTile test suite', () => {
		it('should render', () => {
			const tree = shallow(<WallTile/>)
			expect(tree).toMatchSnapshot()
		})
	})
})