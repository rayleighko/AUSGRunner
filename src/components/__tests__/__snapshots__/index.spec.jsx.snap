// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Layout Components test suite should render 1`] = `ShallowWrapper {}`;

exports[`Components PlayerList Components test suite should render 1`] = `ShallowWrapper {}`;

exports[`Components PlayerList Components test suite should render 2`] = `ShallowWrapper {}`;

exports[`Components Runner Components test suite should render 1`] = `ShallowWrapper {}`;
